src/time/wasm.rs

```diff
@@ -0,0 +1,23 @@
+#[cfg(target_family = "wasm")]
+pub struct Instant {
+    start: f64,
+}
+
+#[cfg(target_family = "wasm")]
+impl Instant {
+    pub fn now() -> Self {
+        let now = js_sys::Date::new_0().get_time();
+        Self { start: now }
+    }
+
+    pub fn elapsed(&self) -> core::time::Duration {
+        let start: f64 = self.start;
+        let now = js_sys::Date::new_0().get_time();
+
+        let mut elapsed = now - start;
+        if elapsed < 0. {
```

Comment:
why would this happen? is the `Date` clock not monotonic?

Reply:
Hello @sarah-ek . Being able to predict the time exactly in javascript is a security vulnerability. Because of that, the javascript api includes a small randomization to the precision of the time. That randomization may cause the calculation of time to become negative. This is why it is checking if the elapsed time is negative.

Reply:
that makes sense. thanks for the explanation